tests/test_operator.py

```diff
@@ -2051,3 +2061,207 @@ def test_indirection(self):
 
         assert np.all(f.data[0] == 0.)
         assert np.all(f.data[i] == 3. for i in range(1, 10))
+
+
+class TestEstimateMemory:
+    """Tests for the Operator.estimate_memory() utility"""
+
+    _array_temp = "r0L0(x, y)" if "CXX" in configuration['language'] else "r0[x][y]"
```

Comment:
You don't need those. Since you only check "in str(op)" all you need is `r0[` which will always be in it either through the Macro def or the indexing.  Same for `_devicelangs` all you need is `isinstance(configuration['platform'], Device)`.  That makes the only two "class" attribute independent of the class that you can just make utilities at the top to avoid inheritance and that odd import try/except